core/src/main/java/hudson/search/Search.java

```diff
@@ -76,7 +76,7 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp) throws IOException,
                     SuggestedItem target = find(index, query, smo);
                     if(target!=null) {
                         // found
-                        rsp.sendRedirect2(a.getUrl()+target.getUrl());
+                        rsp.sendRedirect2(req.getContextPath()+target.getUrl());
```

Comment:
I guess you need to do something like:

```
 Callable<Page> doIndex = new Callable<Page>() {
            @Override public Page call() throws Exception {
                j.jenkins.getSearch().doIndex(Stapler.getCurrentRequest(), Stapler.getCurrentResponse());
                return null;
            }
        };
        Page result = wc.executeOnServer(doIndex);
```
But somehow cusotmize `Stapler.getCurrentRequest(), Stapler.getCurrentResponse()`